docs/src/catalyst_applications/jump_simulation_performance.md

```diff
@@ -0,0 +1,163 @@
+
+#### Unnecessarily putting species in rates rather than in reactions
+Sometimes, a rate has been made dependent on a species, where that species instead could have been made part of the actual reaction. Consider a reaction where an enzyme ($E$) catalyses the phosphorylation of a protein ($X$) to phosphorylated form ($Xᵖ$). It can be written in two different forms:
+```@example jump_simulation_performance_1
+r1 = @reaction k*E, X --> Xᵖ 
+r2 = @reaction k, X + E --> Xᵖ + E
+nothing # hide
+```
+These two reactions will generate identical simulations (this holds for ODE, SDE, and Jump simulations). However, while `r1` will generate a `ConstantRateJump`, `r2` will generate a `MassActionJump`. Hence, if the `r2` form is used, jump simulation performance is, at no cost, improved. Since the two forms are otherwise identical, it is always preferable to, whenever possible, put species in the reactions, rather than the rates.
+
+#### Using piecewise constant, time-dependant, function instead of callbacks.
+Let us consider a reaction with some input that depends on time. We assume that the input is initially $0$, but after some critical time $t=5.0$ it is increased to $1$. This can be implemented through a step function:
+```@example jump_simulation_performance_1
+input(t) = t > 5.0
+r = @reaction input(t), 0 --> X
+nothing # hide
+```
+Here, the production of species $X$ is switched on at the time $t=5.0$. This reaction (which rate depends on `t`) will generate a `VariableRateJump`, which is bad for jump simulation performance. However, since the rate is piecewise constant, it can instead be implemented by setting it to a constant parameter $i$, and then use a [*callback*](@ref advanced_simulations_callbacks) to update it at the critical times. This will again generate an equivalent model, but with the reaction encoded as a `MassActionJump` (rather than a `VariableRateJump`).
+
+## Jump solver selection
+When creating a `JumpProblem`, a specific solver is designated using its third argument.
+```@example jump_simulation_performance_2
+using Catalyst, JumpProcesses, Plots
+
+rn = @reaction_network begin
+    (p,d), 0 <--> X
+end
+u0 = [:X => 10]
+tspan = (0.0, 1000.0)
+ps = [:p => 1000.0, :d => 100.0]
+
+dprob = DiscreteProblem(rn, u0, tspan, ps)
+jprob = JumpProblem(rn, dprob, Direct())
+nothing # hide
+```
+Here (as throughout most of Catalyst's documentation) we have used the `Direct()` solver (which corresponds to Gillespie's original direct method [^2][^3], also called the *stochastic simulation algorithm*). This method was originally published in 1976, and since then, many additional methods for performing jump simulations of CRN models have been developed. 
```

Comment:
```suggestion
Here (as throughout most of Catalyst's documentation) we have used the `Direct()` SSA solver (which corresponds to Gillespie's original direct method [^2][^3]). This method was originally published in 1976, and since then, many additional methods for simulating stochastic chemical kinetics models have been developed. 
```